cpg-language-python/src/main/kotlin/de/fraunhofer/aisec/cpg/frontends/python/StatementHandler.kt

```diff
@@ -615,12 +620,40 @@ class StatementHandler(frontend: PythonLanguageFrontend) :
         return declStmt
     }
 
+    private fun addExternallyImportedToAnalysis(importName: Name) {
+        ctx?.let { ctx ->
+            var currentName = importName
+            while (currentName.isNotEmpty()) {
+                var importPath = currentName.toString().replace(language.namespaceDelimiter, "/")
+
+                language.fileExtensions.forEach { fileExtension ->
+
+                    // Includes a file in the analysis, if it has the root path and
+                    ctx.externalSources
+                        .firstOrNull {
```

Comment:
why only the first? And why are we not looking at the include paths? The external sources set confuses me a bit.